### ISSUE TITLE
Fix S3240 FP: When using same method on different variables

### ISSUE THREAD

**Issue by csaba-sagi-sonarsource** [SonarSource/sonar-dotnet#4962]
### Description

S3240 raises an issue if the same method is invoked but on different variables.

### Repro steps

```
public static string TestFunction(string key1, string key2)
{
    var dictionary1 = new Dictionary<string, string>();
    var dictionary2 = new Dictionary<string, string>();

    string value;
    if (string.IsNullOrEmpty(key1)) dictionary2.TryGetValue(key2, out value);
    else dictionary1.TryGetValue(key1, out value);

    return value;
}
```

### Expected behavior

Rule should not raise any issue.

### Actual behavior

An issue is raised.

### Known workarounds

-

### Related information

* C#/VB.NET Plugins version: 8.30

**Comment by csaba-sagi-sonarsource** [comment]
Created based on: https://community.sonarsource.com/t/sonar-c-analyzer-8-19-0-28253-suggests-incorrect-change-for-rspec-3240/49661
